pmd-java/src/main/resources/category/java/performance.xml

```diff
@@ -112,6 +112,123 @@ public class Test {
         </example>
     </rule>
 
+    <rule name="AvoidFileInputStream"
+          since="5.8.2"
+          message="Avoid instantiating FileInputStream"
+          language="java"
+          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_performance.html#avoidfileinputstream">
```

Comment:
using `Files.newInputStream` reqiures Java 7, so this should be flagged as `minimumLanguageVersion="1.7"`, or a different workaround be proposed.